text/0000-transaction-status.md

```diff
@@ -0,0 +1,118 @@
+
+Finally, the RPC of a node is the source of this information and it allows querying it in a structured format.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The source of transaction status information is the RPC of a node. Both validating and non-validating nodes should have
+access to this information. However, non-validating nodes might have a slightly stale information and they might discover
+about transaction included in chunk simultaneously with it included in the block. So the status known to non-validating
+might jump several items at once, e.g. from "received" to "included in block", or even to "completed".
+
+The structured information returned by an RPC should be exhaustive and forward compatible to avoid breaking wallet and Near shell
+in the future because of the changes in the structure of this format. Wallet and Near shell, however, should also be forward compatible,
+meaning they should not fail if some field is missing from the structured output of RPC.
+
+To simplify the development RPC `tx_details` should return the maximum exhaustive information. And we do not
+create overloads of this RPC based on verbosity. So the following RPC call `tx_details` with `{"tx_hash": "04ffa4f"}`
+will return the following (notice how we allow incomplete hash in the RPC):
+```json
+[
+  {"ReceivedByNonValidator": {"node_id":  "some_node_id"}},
+  {"ReceivedByChunkProducer":  {
+      "node_id": "some_node_id",
+      "shard_id": "some_shard_id",
+      "details": [
+        {"IncludedInChunk": {
+          "chunk_id": "some_chunk_id",
+          "chunk_index": "some_chunk_index",
+          "receipts": ["receipt_hash"],
+          "logs": ["some_logs"],
+          "result": "some_result"}
+        },
+        {"IncludedInBlock": {"block_id": "some_block_id", "block_index": "some_block_index"}}
+      ]
+      }
+  }
+]
+```
+(Note, this JSON output can be a direct result of serializing an internal Rust struct using `serde_json`)
+
+Near shell then would print the entire JSON, upon calling `near tx_details` and support verbosity in two different ways:
+* `--verbosity` with `full`, `normal` (default), and `minimal`.
+* `--format` (see Future Possibilities).
+-- the Go template, e.g.: `--format='{{range $key, $value := .}}{{(index $key "node_id")}}{{end}}'` prints node ids of the participating nodes.
+This is very similar to docker tools, like `docker inspect` that returns a large JSON object (like network or container config) that can be queried using `--format`.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+If the user does not own the node that they use to query such information from Wallet or Near shell then it is unclear what
+would be the incentive system for such nodes to reply. It is clear that we need some incentive system for non-validating nodes to
```

Comment:
the cut of transaction fee for propagation is generally interesting idea, but requires a lot more economics to work. Saito built their whole protocol around such idea actually. 

There are problems with censorship and centralization in this case - when big nodes who are validators will be "cheaper" to send to (because they will be validators and won't take the cut of re-trasmission) - so people will send to them and they can freely censor what they want. And there will be high geographic grouping (because server around the corner will be way faster to send TX to), which means political censorship will be enforce on those. Now, you still can send to other nodes in the network, if you think the local one censors you. But it's possible that by the fact that central nodes will become such dominant they will actually gain % of the stake and just push out independent nodes.